### PR TITLE
[18 India] Add Commodity Icons and Bug Fixes

### PR DESCRIPTION
**Pull request by EdPovi**
- [x] Branch is derived from the latest `master`
- [n/a] Add the `pins` or `archive_alpha_games` label if this change will break existing games
- [x] Code passes linter with `docker compose exec rack rubocop -a`
- [x] Tests pass cleanly with `docker compose exec rack rake`

## Implementation Notes
- Added commodity icons to hexes, refactor Jewelry to use assignment. 
- Edit IPO Row View to handle empty Rows and Corporate Stock Round
- Update Player Value to include corporation Book Value
- Add ability for corporations to convert Railroad Bonds in Phase IV
- May not visit MUMBAI or NEPAL twice 

### Explanation of Change
- Jewelry is allowed to be on the same hex as other commodities, therefore prior method of adding location name could have issues.  
- Added IPO Shares to cache method for shares, it wasn't storing all shares for action hash. 
- IPO Row needed to also create `CorporateBuyCompany` action
- Gauge Change border "Click" action referenced `hex` from border, but this reference can be `nil` after tile upgrade.  Using `@tile` for reference is more reliable. 
- Fixed bug if there were no bonds in Bank Pool. (nil vs empty list)
- Converting Bonds will also check if Manager has changed. 
- Fix bug if all players pass in selling round
- prevent GC border being deleted on tile upgrade.
- Fixed only allowed one phase IV train.
- Allow sale of Director share but may not sell Manager (0%) share

### CODE REVIEW

File: lib/engine/game/g_18_india/game.rb
Comment (michaeljb) [inline lib/engine/game/g_18_india/game.rb:236]: One note on these `LOGGER` calls, and I think I missed this in a prior PR, but the interpolated string should be passed in a block. This defers evaluating the string, meaning in environments where the string is not going to actually get logged, it never gets evaluated. I put some more info about this in #10946, and if we pull in my custom rubocop plugin you'd be able to have rubocop autocorrect these for you. So consider it optional to change them for now.

```suggestion
          LOGGER.debug { "@corporations.flat_map(&:shares) => #{@corporations.flat_map(&:shares)}" }
```

File: lib/engine/game/g_18_india/game.rb
Comment (michaeljb) [inline lib/engine/game/g_18_india/game.rb:649]: This looks a little odd, maybe this instead?

```suggestion
          Array(bank.companies.find { |c| c.type == :bond })
```

File: lib/engine/game/g_18_india/game.rb
Comment (michaeljb) [inline lib/engine/game/g_18_india/game.rb:795]:
```suggestion
          @log << "#{entity.name} is the new Manager of #{corporation.name}"
```

**Review by michaeljb — CHANGES_REQUESTED** [review]
A few small things to change. Exciting that it's close to alpha ready!

File: lib/engine/game/g_18_india/game.rb
Comment (EdPovi) [reply lib/engine/game/g_18_india/game.rb:236]: Many of these LOGGER statements are just for me to get some state information while I was tracking down a bug.  Or simply verify result of some code. 

How do I pull in your plugin?  I'll gladly auto correct them.

File: lib/engine/game/g_18_india/game.rb
Comment (EdPovi) [reply lib/engine/game/g_18_india/game.rb:649]: Will the Array call remove the `nil` result if there isn't a bond?

File: lib/engine/game/g_18_india/game.rb
Comment (EdPovi) [reply lib/engine/game/g_18_india/game.rb]: I have updated the LOGGER calls in this file.  I can also do the other, or wait to try using the RuboCop to auto fix them.

File: lib/engine/game/g_18_india/game.rb
Comment (michaeljb) [reply lib/engine/game/g_18_india/game.rb]: Yep, `Array(nil)` returns `[]`:

```
> docker-compose exec rack bundle exec rake irb
irb -I lib/ -I assets/app/
irb(main):001:0> Array('nil')
=> ["nil"]
irb(main):002:0> Array(nil)
=> []
```

File: lib/engine/game/g_18_india/game.rb
Comment (michaeljb) [reply lib/engine/game/g_18_india/game.rb]: To pull in the plugin you'd add it to your `Gemfile`, as done here - https://github.com/tobymao/18xx/pull/10946/files#diff-d09ea66f8227784ff4393d88a19836f321c915ae10031d16c93d67e6283ab55f

Then run `bundle install` in your container (or simply kill the container and run `make dev_up_b` to rebuild it)

I think it's fine to wait until that's merged in. Thanks for updating the calls here!

**Review by michaeljb — APPROVED** [review]

**Pull request merged**
